test/Dialect/SV/hw-memsim.mlir

```diff
@@ -124,3 +143,200 @@ hw.module.generated @FIRRTLMemTwoAlways, @FIRRTLMem( %wo_addr_0: i4, %wo_en_0: i
+// CHECK-NEXT:    %11 = sv.read_inout %10 : !hw.inout<i4>
+// CHECK-NEXT:    %12 = sv.array_index_inout %Memory2[%7] : !hw.inout<uarray<10xi4>>, i4
+// CHECK-NEXT:    %13 = sv.read_inout %12 : !hw.inout<i4>
+// CHECK-NEXT:    %14 = sv.array_index_inout %Memory3[%7] : !hw.inout<uarray<10xi4>>, i4
+// CHECK-NEXT:    %15 = sv.read_inout %14 : !hw.inout<i4>
+// CHECK-NEXT:    %16 = comb.concat %9, %11, %13, %15 : i4, i4, i4, i4
+// CHECK:   sv.alwaysff(posedge %rw_clock_0)  {
+// CHECK:     sv.passign %42, %rw_wmask_0 : i4
+// CHECK:   }
+// CHECK:   %43 = sv.read_inout %42 : !hw.inout<i4>
+// CHECK:   %44 = sv.reg  : !hw.inout<i4>
+// CHECK:   sv.alwaysff(posedge %rw_clock_0)  {
+// CHECK:     sv.passign %44, %43 : i4
+// CHECK:   }
+// CHECK:   %45 = sv.read_inout %44 : !hw.inout<i4>
+// CHECK:   %46 = sv.reg  : !hw.inout<i4>
+// CHECK:   sv.alwaysff(posedge %rw_clock_0)  {
+// CHECK:     sv.passign %46, %45 : i4
+// CHECK:   }
+// CHECK:   %47 = sv.read_inout %46 : !hw.inout<i4>
+// CHECK:   %48 = comb.extract %47 from 0 : (i4) -> i1
+// CHECK:   %49 = comb.extract %41 from 0 : (i16) -> i4
+// CHECK:   %50 = comb.extract %47 from 1 : (i4) -> i1
+// CHECK:   %51 = comb.extract %41 from 4 : (i16) -> i4
+// CHECK:   %52 = comb.extract %47 from 2 : (i4) -> i1
+// CHECK:   %53 = comb.extract %41 from 8 : (i16) -> i4
+// CHECK:   %54 = comb.extract %47 from 3 : (i4) -> i1
+// CHECK:   %55 = comb.extract %41 from 12 : (i16) -> i4
+// CHECK:   %60 = sv.array_index_inout %Memory0[%23] : !hw.inout<uarray<10xi4>>, i4
+// CHECK:   %61 = sv.read_inout %60 : !hw.inout<i4>
+// CHECK:   %62 = sv.array_index_inout %Memory1[%23] : !hw.inout<uarray<10xi4>>, i4
+// CHECK:   %63 = sv.read_inout %62 : !hw.inout<i4>
+// CHECK:   %64 = sv.array_index_inout %Memory2[%23] : !hw.inout<uarray<10xi4>>, i4
+// CHECK:   %65 = sv.read_inout %64 : !hw.inout<i4>
+// CHECK:   %66 = sv.array_index_inout %Memory3[%23] : !hw.inout<uarray<10xi4>>, i4
+// CHECK:   %67 = sv.read_inout %66 : !hw.inout<i4>
+// CHECK:   %68 = comb.concat %61, %63, %65, %67 : i4, i4, i4, i4
+// CHECK:   %94 = comb.extract %93 from 0 : (i4) -> i1
+// CHECK:   %95 = comb.extract %87 from 0 : (i16) -> i4
+// CHECK:   %96 = comb.extract %93 from 1 : (i4) -> i1
+// CHECK:   %97 = comb.extract %87 from 4 : (i16) -> i4
+// CHECK:   %98 = comb.extract %93 from 2 : (i4) -> i1
+// CHECK:   %99 = comb.extract %87 from 8 : (i16) -> i4
+// CHECK:   %100 = comb.extract %93 from 3 : (i4) -> i1
+// CHECK:   %101 = comb.extract %87 from 12 : (i16) -> i4
+// CHECK:   sv.alwaysff(posedge %wo_clock_0)  {
+// CHECK:     %102 = comb.and %81, %94 : i1
+// CHECK:     sv.if %102  {
+// CHECK:       %106 = sv.array_index_inout %Memory0[%75] : !hw.inout<uarray<10xi4>>, i4
+// CHECK:       sv.passign %106, %95 : i4
+// CHECK:     }
+// CHECK:     %103 = comb.and %81, %96 : i1
+// CHECK:     sv.if %103  {
+// CHECK:       %106 = sv.array_index_inout %Memory1[%75] : !hw.inout<uarray<10xi4>>, i4
+// CHECK:       sv.passign %106, %97 : i4
```

Comment:
This test is going to be very fragile and painful to maintain as syntax and other things shift around.  Please trim this down to the essential check lines.  The examples above are better but can also probably be shrunk a bit.